indico/modules/events/registration/fields/simple.py

```diff
@@ -395,7 +395,7 @@ def _indico_email(value):
 class PictureField(FileField):
     name = 'picture'
     setup_schema_fields = {
-        'min_picture_size': fields.Integer(load_default=0, validate=validate.Range(0, 1200)),
+        'min_picture_size': fields.Integer(load_default=0, validate=validate.Range(0, 1000)),
     }
 
     def get_validators(self, existing_registration):
```

Comment:
Since you convert to JPEG, I think you can change the validation below to only accept JPEG as well. And in principle you need to check that it has the proper size here since nothing would stop you from using a different file upload endpoint to upload a file and then passing its UUID when submitting the registration form.

Alternatively, you could use the file's `meta` field to do something like this:

```python
file.meta['registration_picture_checked'] = True
flag_modified(file, 'meta')
```

And then, in the validation below, just refuse any file that does not have this key.

If you want to go for this option, I suggest adding this to the `UploadFileMixin` so you can easily provide that metadata in your subclass:

```patch
diff --git a/indico/modules/files/controllers.py b/indico/modules/files/controllers.py
index b8773807ef..a7c372e7c1 100644
--- a/indico/modules/files/controllers.py
+++ b/indico/modules/files/controllers.py
@@ -10,6 +10,7 @@ from uuid import UUID

 from flask import request
 from marshmallow import fields
+from sqlalchemy.orm.attributes import flag_modified
 from werkzeug.exceptions import Forbidden, UnprocessableEntity

 from indico.modules.files import logger
@@ -40,6 +41,9 @@ class UploadFileMixin:
         context = self.get_file_context()
         content_type = mimetypes.guess_type(file.filename)[0] or file.mimetype or 'application/octet-stream'
         f = File.create_from_stream(stream, file.filename, content_type, context)
+        if meta := self.get_file_metadata():
+            f.meta.update(meta)
+            flag_modified(f, 'meta')
         return FileSchema().jsonify(f), 201

     def get_file_context(self):
@@ -53,6 +57,13 @@ class UploadFileMixin:
         """
         raise NotImplementedError

+    def get_file_metadata(self):
+        """Get a dict of extra metadata to store on the File.
+
+        This is usually not needed, but could be used e.g. to add some flag which
+        is then used in a validator of the form accepting the file uploaded here.
+        """
+
     def validate_file(self, file):
         """Validate the uploaded file.

```

(as usual, this is untested!)

Reply:
Don't you think that this is unnecessarily restrictive? Why limiting the uploads to JPEG only if it's trivial for Indico to convert PNG (and other image types) to JPEG?

Reply:
I think you misunderstand: We do the conversion during upload. So when sending the file UUID it will always belong to a JPEG file, unless you do funny things w/ other upload endpoints + send a UUID from such an endpoint.